mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BucketNameGenerator.java

```diff
@@ -20,24 +20,62 @@
 package org.apache.james.vault.blob;
 
 import java.time.Clock;
+import java.time.LocalTime;
 import java.time.ZonedDateTime;
+import java.time.temporal.ChronoField;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import javax.inject.Inject;
 
 import org.apache.james.blob.api.BucketName;
 
+import com.google.common.annotations.VisibleForTesting;
+
 public class BucketNameGenerator {
+
+    private static final Pattern BUCKET_NAME_PATTERN = Pattern.compile("deleted-messages-([0-9]{4})-([0-9]{2})-(01)");
+    private static final String BUCKET_NAME_GENRATING_FORMAT = "deleted-messages-%d-%02d-01";
+    private static final int FIRST_DAY_OF_MONTH = 1;
+    private static final int ZERO_HOUR = 0;
+    private static final int ZERO_MINUTE = 0;
+    private static final int ZERO_SECOND = 0;
+    private static final int ZERO_NANOSECOND = 0;
+
     private final Clock clock;
 
     @Inject
-    public BucketNameGenerator(Clock clock) {
+    BucketNameGenerator(Clock clock) {
         this.clock = clock;
     }
 
-    public BucketName currentBucket() {
+    BucketName currentBucket() {
         ZonedDateTime now = ZonedDateTime.now(clock);
-        int month = now.getMonthValue();
+        int month = now.getMonth().get(ChronoField.MONTH_OF_YEAR);
         int year = now.getYear();
-        return BucketName.of(String.format("deleted-messages-%d-%02d-01", year, month));
+        return BucketName.of(String.format(BUCKET_NAME_GENRATING_FORMAT, year, month));
+    }
+
+    @VisibleForTesting
+    boolean isBucketRangeBefore(BucketName bucketName, ZonedDateTime beforeThisTime) {
```

Comment:
IMO you can simply return the ZoneDateTime that this bucket started being used at, and let the caller do the comparison (beforeThisTime parameter should be removed IMO)